Bugfix/remove reward metric key

danoswaltCL opened the pull request:
https://github.com/CarnegieLearningWeb/UpGrade/issues/2741

Removes Reward Metric views from UI
Removes unique name constraints on Mooclet Experiments
Removes /log based reward system and all tie-ins to upgrade metrics systems
Removes the automatic metric query creation
Adds a /reward endpoint to client controller, which will lookup experiment info and do a more direct call to mooclet
Adds a sendReward() endpoint to TS client lib
Updates quicktest to add rewards and some more readable errors

danoswaltCL committed:
provisional changes before adding js client update

danoswaltCL committed:
Merge remote-tracking branch 'origin/release/6.2' into bugfix/remove-reward-metric-key

danoswaltCL committed:
client updates

danoswaltCL committed:
tests and fix error logging

danoswaltCL committed:
update swagger examples

danoswaltCL committed:
add comment about pooling logic

danoswaltCL committed:
update version and package-locks

danoswaltCL committed:
kick cicd

danoswaltCL committed:
kick cicd

danoswaltCL committed:
kick cicd

danoswaltCL committed:
kick cicd

danoswaltCL committed:
fix typos and test error usage

danoswaltCL committed:
fix types path

bcb37 reviewed (commented):
This looks ok to me and seems to work. Just two thoughts:

1. The `mooclet_experiment_ref` table still has the (now unused) `rewardMetricKey` field. That's probably not going to get in the way, but it could be confusing in the future - although it's not the only leftover in the database schema.
2. The only way to see reward data in the UpGrade UI is now in the policy parameters JSON listing, but the reward values in that view don't seem to get updated when a new reward value is sent until an /assign call is made, even though it's there in the mocclet db. Could be misleading if we care about displaying that information accurately.

danoswaltCL commented:
1. true, should i add a migration for that? seems pretty safe to me.
2. yes, I'm not sure what to do about that right at the moment. currentPosteriors is prob not the right source, for # of rewards per condition we'll prob be able query the mooclet db directly, that info is in there in the `value` table. i think that will need to be for later though.

@bcb37

danoswaltCL commented on backend/packages/Upgrade/src/api/services/MoocletRewardsService.ts:
```diff
@@ -1,227 +1,204 @@
 export class MoocletRewardsService {
+  userId: string;
+  request: RewardValidator;
+  logger: UpgradeLogger;
```
hm let me look at this...

landaetaluis committed:
NO-TICKET misc/upgrade_client_lib: Move default entrypoint types to root (#2779)

- This will help other (non TS, less smart) tools, like Jest, to know where to look up for types

danoswaltCL committed:
removes the class properties in moocletrewardservice, need to pass through functions

danoswaltCL committed:
bump version

danoswaltCL committed:
Merge branch 'release/6.2' into bugfix/remove-reward-metric-key

danoswaltCL committed:
fix wrong comment

bcb37 commented:
> 1. true, should i add a migration for that? seems pretty safe to me.
>
Nah, probably safer for now to leave it, since there might be data for existing experiments on one of the environments, which we'll just ignore.

> 2. yes, I'm not sure what to do about that right at the moment. currentPosteriors is prob not the right source, for # of rewards per condition we'll prob be able query the mooclet db directly, that info is in there in the `value` table. i think that will need to be for later though.
> 
Ok makes sense.

> @bcb37
